tests/cron.go

```diff
@@ -561,42 +568,63 @@ func (s *ClientFunctionalSuite) TestCronWorkflowCompletionStates() {
 
 func (s *ClientFunctionalSuite) listOpenWorkflowExecutions(start, end time.Time, id string, expectedNumber int) []*workflowpb.WorkflowExecutionInfo {
 	s.T().Helper()
-	for i := 0; i < 20; i++ {
-		resp, err := s.sdkClient.ListOpenWorkflow(NewContext(), &workflowservice.ListOpenWorkflowExecutionsRequest{
-			Namespace:       s.namespace,
-			MaximumPageSize: int32(2 * expectedNumber),
-			StartTimeFilter: &filterpb.StartTimeFilter{EarliestTime: timestamppb.New(start), LatestTime: timestamppb.New(end)},
-			Filters: &workflowservice.ListOpenWorkflowExecutionsRequest_ExecutionFilter{ExecutionFilter: &filterpb.WorkflowExecutionFilter{
-				WorkflowId: id,
-			}},
-		})
-		s.NoError(err)
-		if len(resp.GetExecutions()) == expectedNumber {
-			return resp.GetExecutions()
-		}
-		time.Sleep(250 * time.Millisecond)
-	}
-	s.FailNow("didn't get expected number")
-	panic("unreached")
+	var resp *workflowservice.ListOpenWorkflowExecutionsResponse
+	s.Eventuallyf(
+		func() bool {
+			var err error
+			resp, err = s.sdkClient.ListOpenWorkflow(
+				NewContext(), &workflowservice.ListOpenWorkflowExecutionsRequest{
+					Namespace:       s.namespace,
+					MaximumPageSize: int32(2 * expectedNumber),
+					StartTimeFilter: &filterpb.StartTimeFilter{
+						EarliestTime: timestamppb.New(start),
+						LatestTime:   timestamppb.New(end),
+					},
+					Filters: &workflowservice.ListOpenWorkflowExecutionsRequest_ExecutionFilter{
+						ExecutionFilter: &filterpb.WorkflowExecutionFilter{
+							WorkflowId: id,
+						},
+					},
+				},
+			)
+			s.NoError(err)
+			return len(resp.GetExecutions()) == expectedNumber
+		},
+		waitForESToSettle,
+		100*time.Millisecond,
+		"didn't get expected number",
```

Comment:
Expected number of what? Could you be more descriptive here, like "Found %d executions but expected %d"?